src/test/llmnode_test.cpp

```diff
@@ -2809,6 +2810,303 @@ TEST_F(EmbeddingsExtensionTest, simplePositive) {
+    ASSERT_EQ(embeddingsRequest.encoding_format, EncodingFormat::FLOAT);
+    auto strings = std::get_if<std::vector<std::string>>(&embeddingsRequest.input);
+    ASSERT_NE(strings, nullptr);
+    ASSERT_EQ(strings->size(), 3);
+    ASSERT_EQ(strings->at(0), "one");
+    ASSERT_EQ(strings->at(1), "two");
+    ASSERT_EQ(strings->at(2), "three");
+}
+
+class EmbeddingsInvalidConfigTest : public ::testing::Test {
+protected:
+    static std::unique_ptr<std::thread> t;
+
+public:
+    std::unique_ptr<ovms::HttpRestApiHandler> handler;
+
+    std::vector<std::pair<std::string, std::string>> headers;
+    ovms::HttpRequestComponents comp;
+    const std::string endpointEmbeddings = "/v3/embeddings";
+    MockedServerRequestInterface writer;
+    std::string response;
+    ovms::HttpResponseComponents responseComponents;
+
+    static void SetUpTestSuite() {
+        std::string port = "9173";
+        ovms::Server& server = ovms::Server::instance();
+        const char* configPath = "/ovms/src/test/embeddings/invalid_config_embeddings.json";
+        server.setShutdownRequest(0);
+        randomizePort(port);
+        char* argv[] = {(char*)"ovms",
+            (char*)"--config_path",
+            (char*)configPath,
+            (char*)"--port",
+            (char*)port.c_str()};
+        int argc = 5;
+        t.reset(new std::thread([&argc, &argv, &server]() {
+            EXPECT_EQ(EXIT_SUCCESS, server.start(argc, argv));
+        }));
+        auto start = std::chrono::high_resolution_clock::now();
+        const int numberOfRetries = 5;
+        while ((server.getModuleState(ovms::SERVABLE_MANAGER_MODULE_NAME) != ovms::ModuleState::INITIALIZED) &&
+               (std::chrono::duration_cast<std::chrono::seconds>(std::chrono::high_resolution_clock::now() - start).count() < numberOfRetries)) {
+        }
+    }
+
+    void SetUp() {
+        ovms::Server& server = ovms::Server::instance();
+        handler = std::make_unique<ovms::HttpRestApiHandler>(server, 5);
+        ASSERT_EQ(handler->parseRequestComponents(comp, "POST", endpointEmbeddings, headers), ovms::StatusCode::OK);
+    }
+
+    static void TearDownTestSuite() {
+        ovms::Server& server = ovms::Server::instance();
+        server.setShutdownRequest(1);
+        t->join();
+        server.setShutdownRequest(0);
+    }
+
+    void TearDown() {
+        handler.reset();
+    }
+};
+std::unique_ptr<std::thread> EmbeddingsInvalidConfigTest::t;
+
+TEST_F(EmbeddingsInvalidConfigTest, simpleNegative) {
+    std::string requestBody = R"(
+        {
+            "model": "embeddings",
+            "input": "dummyInput"
+        }
+    )";
+    Status status = handler->dispatchToProcessor(endpointEmbeddings, requestBody, &response, comp, responseComponents, &writer);
+    ASSERT_EQ(status, ovms::StatusCode::MEDIAPIPE_EXECUTION_ERROR) << status.string();
+}
+
+class EmbeddingsInvalidTokenizerConfigTest : public ::testing::Test {
```

Comment:
You can use inheritence to avoid rewritting entire class for both `EmbeddingsInvalidTokenizerConfigTest` and `EmbeddingsInvalidConfigTest`